lib/tuples.gd

```diff
@@ -178,3 +178,43 @@ direct product elements families" );
 DeclareOperation( "DirectProductElement", [ IsList ]);
 DeclareOperation( "DirectProductElementNC",
     [ IsDirectProductElementFamily, IsList ]);
+
+
+#############################################################################
+##
+##
+##  <#GAPDoc Label="DirectProductFamily">
+##  <ManSection>
+##  <Func Name="DirectProductFamily" Arg='args'/>
+##
+##  <Description>
+##  <A>args</A> must be a dense list of <Ref Attr="CollectionsFamily"/>
+##  families, otherwise the function raises an error.
+##  <P/>
+##  <Ref Func="DirectProductFamily"/> returns <C>fam</C>, a collections
+##  family of <Ref Filt="IsDirectProductElement"/> objects.
+##  <P/>
+##  <C>fam</C> is the <Ref Filt="CollectionsFamily"/> of
+##  <Ref Filt="IsDirectProductElement"/> objects
+##  whose <C>i</C>-th component is in <C>ElementsFamily(args[i])</C>.
+##  <P/>
+##  Note that a collection in <C>fam</C> must thus not necessarily be a
+##  direct product.
+##  Also note that not all direct products in &GAP; are created via these
+##  families. For example the elements of a permutation group created via
+##  <Ref Func="DirectProduct"/> are not <Ref Filt="IsDirectProductElement"/>
+##  objects.
+##  <P/>
+##  <Example><![CDATA[
+##  gap> D8 := DihedralGroup(IsPermGroup, 8);;
+##  gap> FamilyObj(D8) = CollectionsFamily(PermutationsFamily);
+##  true
+##  gap> fam := DirectProductFamily([FamilyObj(D8), FamilyObj(D8)]);;
+##  gap> ComponentsOfDirectProductElementsFamily(ElementsFamily(fam));
+##  [ <Family: "PermutationsFamily">, <Family: "PermutationsFamily"> ]
+##  ]]></Example>
```

Comment:
There's nothing wrong with this example, but as stated elsewhere the direct product of permutation groups is again a permutation group, in the permutations family. In my opinion, it would be nice to _also_ have an example where the family of the resulting direct product is the `DirectProductFamily`, eg. something along the lines of:
```
gap> D8 := DihedralGroup(IsPermGroup, 8);;
gap> D4 := DihedralGroup(IsPcGroup, 4);;
gap> fam := DirectProductFamily([FamilyObj(D8), FamilyObj(D4)]);;
gap> fam = FamilyObj(DirectProduct(D8, D4));
true
```
Is that a reasonable request? Do you think it makes thing less confusing, or more confusing?

Reply:
I think that makes a lot of sense.

Reply:
Ooh, nice. I wasn't aware that you could do that. I'll add that.

I'll put the explanation about the direct product of permutation groups being a permutation group and your example into a new little paragraph. I hope that will make the documentation clearer.